### DISCUSSION TITLE
Issues with Accessing the Livewire JavaScript component instance

### DISCUSSION THREAD

**Discussion by fabianbartsch** [wire-elements/modal#253] [Q&A]
The Livewire documentation explains how to access the JavaScript component: [Accessing the JavaScript component instance](https://laravel-livewire.com/docs/2.x/inline-scripts#accessing-javascript-component-instance)

```html
<script>
    document.addEventListener('livewire:load', function () {
        // Get the value of the "count" property
        var someValue = @this.count
 
        // Set the value of the "count" property
        @this.count = 5
 
        // Call the increment component action
        @this.increment()
 
        // Run a callback when an event ("foo") is emitted from this component
        @this.on('foo', () => {})
    })
</script>
```

But for the life of me I can't get the event listener "livewire:load" to execute, and earlier the attempt to access `@this` throws an error.

Is this a known issue? I'm I missing something here?

**Comment by PhiloNL** [comment]
Are you putting this code inside a modal?

**Reply by fabianbartsch** [reply]
It's part of the blade component that will be rendered by the livewire component which extends the modal class yes.
